statistics/most_distinct_dates_competed_on.rb

```diff
@@ -38,4 +38,34 @@ def query
       ORDER BY attended_dates DESC, person.name
     SQL
   end
+
+  def transform(query_results)
+    query_results.map do |result|
+      dates_list = transform_dates(result["dates_list"])
+      [result["attended_dates"], result["person_link"], dates_list]
+    end
+  end
+
+  def transform_dates(dates_list)
```

Comment:
Ruby has a lot of functions for working with collections, here we can use group by and sort to make it more declarative :)

```ruby
require "date"

# ...

def transform_dates(dates_list)
  dates_list
    .split(',')
    .map { |date| date.split('/').map(&:to_i) }
    .group_by { |month, day| month }
    .sort_by { |month, dates| month }
    .map do |month, dates|
      days = dates.map { |month, day| day }
      "#{Date::MONTHNAMES[month]}: #{days.join(", ")}"
    end
    .join("<br />")
end
```

Instead of using `.map(&:strip)` we can change the SQL query to join with "," rather than ", ".